blog/content/post/2021-02-23-OSF_vs_OCP_OSF.md

```diff
@@ -0,0 +1,119 @@
+| -------------:|:-------------:|:-----------------:|:------------:|:-----------------:|:-------------- |
+| Compute       | $13.25        | 83.1%             | $28.07       | 83.0%             | 16.2%          |
+| Storage       | $2.45         | 16.9%             | $5.73        | 17.0%             | 18.5%          |
+| **Total**         | **$15.70**        | **100.0%**            | **$33.80**       | **100.0%**            | **16.6%**          |
+
+*Source: IDC Worldwide Open Compute Project Compute and Storage Infrastructure Market Forecast, May 2020.*
+
+The OCP Open System Firmware (OSF) project is aiming to allow system firmware
+like [BIOS][3] and [BMC][4] to be modified and their source code shared with
+the owners of OCP-certified hardware. Since March 2021, supporting OSF is
+mandatory for servers with OCP badging. The ecosystem partners have decided to
+implement such open-source firmware as coreboot for their OSF architecture,
+since having the source helps to adapt the firmware to the needs of even the
+most demanding clients by fixing the firmware bugs and expanding the device's
+feature set - at a reduced cost - which allows making the most out of both
+your existing and future products by extending their lifecycle and bringing them
+to the new previously unthought-of markets.
+
+![Embedded software market](/img/Embedded_Software_Market.png)
+
+Just being open-source - by itself is not enough for the firmware to meet the
+OCP certification: the extra requirements are mentioned in a [OSF Checklist][5].
+The source code should be easily buildable by a native (ideally the open-source)
+toolchain and the flashing utility should be either open-source like [flashrom][6]
+or at least documented well enough for an open-source implementation to be
+written. The key OSF components have to be open-source, and although it is
+allowed to include some closed-source ones *(aka binary blobs)* to OSF - there
+should be a valid justification for them to be approved, such as "containing
+silicon IP": i.e. a blob has been provided by your SV (Silicon Vendor) like
+Intel to set up their chip during boot, which is needed for your device to run.
+And all the documentation should be good enough and distributable without NDAs.
+
+![OSF features](/img/OSF_features.png)
+
+The diagram above shows 6 types of OSF features: the fundamental features to the
+more advanced ones are depicted from left to right and 1st to 3rd are essential.
+
+1. Chip Initialization and Integration: the firmware components for the initial
+setup of the motherboard's key chips - like the chips themselves - are usually
+provided by the Silicon Vendors, sometimes in a binary-only form *(without the*
+*source code)* like Intel FSP and later AMD AGESA. Some binary-only components,
+depending on their complexity, could be reverse-engineered by us at [3mdeb][7] to
+create a higher quality opensource replacement with less bugs and more features.
```

Comment:
@mikebdp2 this reply on Reddit show other paths: https://www.reddit.com/r/3mdeb/comments/lf9z0v/qa_post/gn0dj81/?utm_source=reddit&utm_medium=web2x&context=3

Reply:
@pietrushnic Done, added more information of 3mdeb services to this place.